### ISSUE TITLE
Debug RNN

### ISSUE THREAD

**Issue by michaelchughes** [tufts-ml/time_series_prediction#11]
start by looking at RNN and why the area-under-ROC (AUROC) is only a small bit better than random (0.5).
Here are some questions to help with debugging:
• what is the final AUROC on the training and validation set? this will help diagnose over-vs-under fitting
• even better, can we make plots of AUROC over time for train/valid/test?
(x axis: epoch of training, starting from 0, y-axis: AUROC, show different lines for train/valid/test)?
• how many states does the RNN have? how many total scalar parameters that need to be learned?

Then, you have two choices:
-- if you observe underfitting (train AUROC is near 0.5, not improving much), would increasing the number of parameters help?
-- if you observe overfitting, does decreasing parameters or adding regularization help?

**Comment by michaelchughes** [comment]
Gonna mark this as resolved, since we've done a lot on this in the spring.
